### ISSUE TITLE
unused_variables: false positive with REDUCE

### ISSUE THREAD

**Issue by nununo** [abaplint/abaplint#2061]
I believe this REDUCE yielded two false positives (`value_request` and `value`) in [here](https://github.com/abapinho/abapTimeMachine/pull/95/checks?check_run_id=3089054076):

```abap
  METHOD get_request_count_for_value.
    result = REDUCE int2(
      INIT x = 0
      FOR value_request IN value_requests
      WHERE ( value = value )
      NEXT x = x + 1 ).
  ENDMETHOD.
```

- `value_request` is indeed not used but the syntax requires it
- `value` is used

**Comment by larshp** [comment]
reproduced as

```abap
TYPES: BEGIN OF ty_bar,
         value TYPE i,
       END OF ty_bar.
DATA value_requests TYPE STANDARD TABLE OF ty_bar WITH EMPTY KEY.
DATA value TYPE i.

DATA(result) = REDUCE int2(
  INIT x = 0
  FOR value_request IN value_requests
  WHERE ( value = value )
  NEXT x = x + 1 ).
WRITE result.
```

![image](https://user-images.githubusercontent.com/5888506/126027871-333626d6-a554-4130-b8c5-054e1f2e0404.png)

**Comment by larshp** [comment]
@nununo try again, latest version

abaplint.app updated, just push another commit to your repo

**Comment by larshp** [comment]
zero issues found, https://github.com/abapinho/abapTimeMachine

**Comment by nununo** [comment]
I was gonna ping you about it 😄 but you found it already. Congratulations 😉
